lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +557,27 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      out, err, st = kubectl.run("get", "secret", EjsonSecretProvisioner::EJSON_KEYS_SECRET,
+        output: "json", output_is_sensitive: true)
+      unless st.success? || err.include?(KubernetesDeploy::Kubectl::NOT_FOUND_ERROR_TEXT)
+        raise FatalDeploymentError,
+          "Error running validation for Secret/#{EjsonSecretProvisioner::EJSON_KEYS_SECRET}: #{err}"
+      end
+
+      if st.success?
+        secret = JSON.parse(out)
+        return unless secret.dig("metadata", "annotations", KubernetesResource::LAST_APPLIED_ANNOTATION)
+        @logger.error("Deploy cannot proceed because protected resource " \
+          "Secret/#{EjsonSecretProvisioner::EJSON_KEYS_SECRET} would be pruned.")
+
+        raise EjsonPrunableError.exception("Found #{KubernetesResource::LAST_APPLIED_ANNOTATION} annotation on " \
```

Comment:
I asked this previously but maybe you missed it: why `.exception`? It seems to do the exact same thing as `.new` but I had never seen it before and had to look it up.